Minor issue to build dict-benchmark

Hi, I tried to build dict-benchmark and saw some error messages because some references like jemalloc_XXX functions are missed.

I guess this is a minor issue but the current makefile incurs this problem.
At line 217 and 218 of src/Makefile,

dict-benchmark: dict.c zmalloc.c sds.c
	$(REDIS_CC) $(FINAL_CFLAGS) dict.c zmalloc.c sds.c siphash.c -D DICT_BENCHMARK_MAIN -o dict-benchmark

could be changed such as

dict-benchmark: dict.c zmalloc.c sds.c
     $(REDIS_CC) $(FINAL_CFLAGS) $^ -D DICT_BENCHMARK_MAIN -o $@ $(FINAL_LIBS)

Thanks,
Jaebaek

I guess the reason why you did not meet the problem is because redis sets libc's malloc as default on OSX.
In Linux, redis sets jemalloc as the default.

By the way,

$ uname -a

showed me

Linux jaebaek-sgx 4.4.0-72-generic #93-Ubuntu SMP Fri Mar 31 14:07:41 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

I installed Ubuntu 16.04

I tried

$ make
$ cd src
$ make dict-benchmark

and it showed me

    CC Makefile.dep
    CC dict-benchmark
/tmp/ccvBGxxm.o: In function `zmalloc':
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:107: undefined reference to `je_malloc'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:111: undefined reference to `je_malloc_usable_size'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:111: undefined reference to `je_malloc_usable_size'
/tmp/ccvBGxxm.o: In function `zmalloc_no_tcache':
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:125: undefined reference to `je_mallocx'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:127: undefined reference to `je_malloc_usable_size'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:127: undefined reference to `je_malloc_usable_size'
/tmp/ccvBGxxm.o: In function `zfree_no_tcache':
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:133: undefined reference to `je_malloc_usable_size'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:133: undefined reference to `je_malloc_usable_size'
/tmp/ccvBGxxm.o: In function `zcalloc':
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:139: undefined reference to `je_calloc'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:143: undefined reference to `je_malloc_usable_size'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:143: undefined reference to `je_malloc_usable_size'
/tmp/ccvBGxxm.o: In function `zrealloc':
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:161: undefined reference to `je_malloc_usable_size'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:162: undefined reference to `je_realloc'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:166: undefined reference to `je_malloc_usable_size'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:166: undefined reference to `je_malloc_usable_size'
/tmp/ccvBGxxm.o: In function `zfree':
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:203: undefined reference to `je_malloc_usable_size'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:203: undefined reference to `je_malloc_usable_size'
/tmp/ccvBGxxm.o: In function `zfree_no_tcache':
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:134: undefined reference to `je_dallocx'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:134: undefined reference to `je_dallocx'
/tmp/ccvBGxxm.o: In function `zfree':
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:204: undefined reference to `je_free'
/home/jaebaek/sgx-mutual-untrust/graphene-and-sdk/redis/src/zmalloc.c:204: undefined reference to `je_free'
collect2: error: ld returned 1 exit status
Makefile:218: recipe for target 'dict-benchmark' failed
make: *** [dict-benchmark] Error 1

--- a/src/Makefile
+++ b/src/Makefile
@@ -214,8 +214,8 @@ $(REDIS_BENCHMARK_NAME): $(REDIS_BENCHMARK_OBJ)
 $(REDIS_CHECK_AOF_NAME): $(REDIS_CHECK_AOF_OBJ)
 	$(REDIS_LD) -o $@ $^ $(FINAL_LIBS)
 
-dict-benchmark: dict.c zmalloc.c sds.c
-	$(REDIS_CC) $(FINAL_CFLAGS) dict.c zmalloc.c sds.c siphash.c -D DICT_BENCHMARK_MAIN -o dict-benchmark
+dict-benchmark: dict.c zmalloc.c sds.c siphash.c
+	$(REDIS_CC) $(FINAL_CFLAGS) $^ -D DICT_BENCHMARK_MAIN -o $@ $(FINAL_LIBS)
 
 # Because the jemalloc.h header is generated as a part of the jemalloc build,
 # building it should complete before building any other object. Instead of

